tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="DatabaseMonitoringPropagator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.DatabaseMonitoring
+{
+    internal static class DatabaseMonitoringPropagator
+    {
+        private const string SqlCommentSpanService = "dddbs";
+        private const string SqlCommentRootService = "ddps";
+        private const string SqlCommentVersion = "ddpv";
+        private const string SqlCommentEnv = "dde";
+
+        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, Span span)
+        {
+            if (propagationStyle == DbmPropagationLevel.Disabled)
+            {
+                return string.Empty;
+            }
+
+            var propagatorSringBuilder = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+            propagatorSringBuilder.Append($"/*{SqlCommentRootService}='{configuredServiceName}',{SqlCommentSpanService}='{span.ServiceName}'");
+
+            if (span.GetTag(Tags.Version) is { } versionTag)
```

Comment:
```suggestion
            if (span.Context.TraceContext?.ServiceVersion) is { } versionTag)
```

Reply:
This is what we do internally so you can just skip `GetTag()` and go directly to the source. It's only there as a semi-public API. Same for `Env` below.
https://github.com/DataDog/dd-trace-dotnet/blob/806efbb055a9cf54a4be1b4a8c0a69049de6a973/tracer/src/Datadog.Trace/Span.cs#L392-L395